doc/usage.md

```diff
@@ -0,0 +1,69 @@
+## Usage
+
+This document explains how you can build your enclave application on top of
+nitriding.  The following steps are necessary.
+
+1. Make sure that your enclave application supports [reproducible
+   builds](https://reproducible-builds.org); otherwise, users won't be able to
+   verify your enclave image.  Both Rust and Go support reproducible builds
+   although some effort may be necessary to get there.
+   [Nitriding's Makefile](../cmd/Makefile) shows how one can build a Go program
+   reproducibly.
+
+2. Set up
+   [this proxy application](https://github.com/containers/gvisor-tap-vsock/tree/main/cmd/gvproxy)
+   on the EC2 host.  Run it as follows:
+   ```
+   gvproxy -listen vsock://:1024 -listen unix:///tmp/network.sock
+   ```
+   Next, tell the proxy application to forward port 443 to nitriding.
+   ```
+   curl \
+     --unix-socket /tmp/network.sock \
+     http:/unix/services/forwarder/expose \
+     -X POST \
+     -d '{"local":":443","remote":"192.168.127.2:443"}'
+   ```
+   In case you're wondering, 192.168.127.2 is nitriding's static IP address in
+   the private network between gvproxy and nitriding.  Does your enclave
+   application expose any other ports?  If so, you have to forward these ports
+   too.
+
+3. Build the nitriding executable by running `make cmd/nitriding`.
+   (Then, run `./cmd/nitriding -help` to see a list of command line options.)
+   For reproducible Docker images, we recommend
+   [kaniko](https://github.com/GoogleContainerTools/kaniko)
+   or
+   [ko](https://github.com/ko-build/ko) (for Go applications only).
+   Take a look at [this
+   Makefile](https://github.com/brave/star-randsrv/blob/main/Makefile) to see
+   an application of kaniko.
+
+3. Bundle the freshly-compiled nitriding and your enclave application together
+   in a Dockerfile.  The nitriding stand-alone executable must be invoked
```

Comment:
s/in/with/

The applications are bundled in the container image, which is described by the Dockerfile.

Reply:
Addressed in b472678.